### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by otraore**
Also fixes https://github.com/EngoEngine/engo/issues/282

### CODE REVIEW

**Comment by paked** [comment]
> audio_unsupported.go:6:2: "log" imported but not used

**Comment by EtienneBruines** [comment]
Once it builds, it LGTM.

File: shaders.go [hunk 1]
```diff
@@ -646,7 +645,7 @@ func (l *legacyShader) generateBufferContent(ren *RenderComponent, space *SpaceC
 			}
 		}
 	default:
-		log.Println("Warning: type not supported")
+		unsupportedType()
```
Comment (mewmew) [inline shaders.go]:
```go
unsupportedType(shape)
```

See updated function definition of `unsupportedType` below.

File: shaders.go [hunk 2]
```diff
@@ -731,7 +730,7 @@ func (l *legacyShader) Draw(ren *RenderComponent, space *SpaceComponent) {
 			Gl.DrawArrays(Gl.LINE_LOOP, len(shape.Points), len(shape.Points))
 		}
 	default:
-		log.Println("Warning: type not supported")
+		unsupportedType()
```
Comment (mewmew) [inline shaders.go]:
```go
unsupportedType(shape)
```

See updated function definition of `unsupportedType` below.

File: utils.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package engo
+
+import "log"
+
+func notImplemented(msg string) {
+	warning(msg + "is not yet implemented on this platform")
+}
+
+func unsupportedType() {
+	warning("type not supported")
```
Comment (mewmew) [inline utils.go]:
```go
func unsupportedType(v interface{}) {
   warning("type %T not supported", v)
}
```

File: utils.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+func warning(msg string) {
```
Comment (mewmew) [inline utils.go]:
```go
func warning(format string, a ...interface{}) {
   log.Printf("[WARNING] " + format, a...)
}
```

File: utils.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package engo
+
+import "log"
+
+func notImplemented(msg string) {
+	warning(msg + "is not yet implemented on this platform")
```
Comment (mewmew) [inline utils.go]: missing space?

```patch
-warning(msg + "is not yet implemented on this platform")
+warning(msg + " is not yet implemented on this platform")
```

**Pull request merged**